firedrake/slate/preconditioners.py

```diff
@@ -149,27 +183,19 @@ def initialize(self, pc):
             Smat.setNullSpace(nullspace)
 
         # Set up the KSP for the system of Lagrange multipliers
-        ksp = PETSc.KSP().create(comm=pc.comm)
-        ksp.setOptionsPrefix(prefix)
-        ksp.setOperators(Smat)
-        ksp.setUp()
-        ksp.setFromOptions()
-        self.ksp = ksp
+        trace_ksp = PETSc.KSP().create(comm=pc.comm)
+        trace_ksp.setOptionsPrefix(prefix)
+        trace_ksp.setOperators(Smat)
+        trace_ksp.setUp()
+        trace_ksp.setFromOptions()
+        self.trace_ksp = trace_ksp
 
         split_mixed_op = dict(split_form(Atilde.form))
         split_trace_op = dict(split_form(K.form))
 
         # Generate reconstruction calls
         self._reconstruction_calls(split_mixed_op, split_trace_op)
 
-        # Set up the projectors
-        data_params = {"ksp_type": "preonly",
-                       "pc_type": "bjacobi",
-                       "sub_pc_type": "ilu"}
-        self.data_projector = Projector(self.unbroken_rhs.split()[self.vidx],
-                                        self.broken_rhs.split()[self.vidx],
-                                        solver_parameters=data_params)
-
         # NOTE: Tolerance is very important here and so we provide
         # the user a way to specify projector tolerance
         opts = PETSc.Options()
```

Comment:
Can we, while we're here, set this projector up as a KSP, so we can control all the options?  Or else, add an `options_prefix` setting to `Projector`/`project`.